x-pack/platform/test/onechat_functional/tests/converse/conversation_history.ts

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import type { LlmProxy } from '../../../onechat_api_integration/utils/llm_proxy';
+import { createLlmProxy } from '../../../onechat_api_integration/utils/llm_proxy';
+import { createConnector, deleteConnectors } from '../../utils/connector_helpers';
+import type { FtrProviderContext } from '../../../functional/ftr_provider_context';
+
+const APP_ID = 'agent_builder';
+const MOCKED_CONVERSATIONS = [
+  {
+    id: 'conv-1',
+    title: 'Conversation1',
+    user_message: 'Hello, this is conversation 1',
+    assistant_response: 'This is the response for conversation 1',
+  },
+  {
+    id: 'conv-2',
+    title: 'Conversation2',
+    user_message: 'Hello, this is conversation 2',
+    assistant_response: 'This is the response for conversation 2',
+  },
+  {
+    id: 'conv-3',
+    title: 'Conversation3',
+    user_message: 'Hello, this is conversation 3',
+    assistant_response: 'This is the response for conversation 3',
+  },
+];
+
+export default function ({ getPageObjects, getService }: FtrProviderContext) {
+  const { common } = getPageObjects(['common']);
+  const testSubjects = getService('testSubjects');
+  const retry = getService('retry');
+  const es = getService('es');
+  const log = getService('log');
+  const supertest = getService('supertest');
+
+  describe('Conversation History', function () {
+    let llmProxy: LlmProxy;
+
+    before(async () => {
+      llmProxy = await createLlmProxy(log);
+      await createConnector(llmProxy, supertest);
+    });
+
+    after(async () => {
+      llmProxy.close();
+      await deleteConnectors(supertest);
+    });
+
+    beforeEach(async () => {
+      const bulkBody = MOCKED_CONVERSATIONS.flatMap((conv) => [
+        { index: { _index: '.chat-conversations', _id: conv.id } },
```

Comment:
NIT: could import `conversationIndexName` from `@kbn/onechat-plugin/server/services/conversation/storage.ts` instead of hardcoding the index name